File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def solution(board, moves):
    answer = 0
    n = len(board)
    m = len(board[0])
    save_item = [] # 아이템을 저장할 배열 생성
    
    for move in moves:
        for i in range(n):
            target = board[i][move - 1]
            if target != 0:
                # 아이템을 저장하는 배열에 아이템이 없다면 append
                if len(save_item) == 0:
                    save_item.append(target)
                else:
                    # 같은 아이템이 배열에 들어오면 터트리기
                    if save_item[-1] == target:
                        save_item.pop()
                        answer += 2
                    # 다른 아이템이 배열에 들어오면 저장하기
                    else:
                        save_item.append(target)
                board[i][move - 1] = 0
                break
                
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0;
    const stack = [];
    const n = board.length;
    moves.forEach((item) => {
        
        for (let i = 0; i < n; i++) {
            const targetItem = board[i][item - 1];
            if (targetItem) {
                if (stack.length) {
                    if (stack[stack.length - 1] === targetItem) {
                        answer += 2;
                        stack.pop();
                    } else {
                        stack.push(targetItem);
                    }
                } else {
                    stack.push(targetItem);
                }
                board[i][item - 1] = 0;
                break;
            }
        }
        
    })
    
    return answer;
}